internal/secrets/azure_keyvault.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 Google LLC
```

Comment:
Do we have a way to test these implementations?

Reply:
We can test the Vault one fairly easily since it's a standard HTTP server. I don't have a good idea of testing the Azure one without uploading Azure keys. We could do it (e.g. store them in Secret Manager and retrieve them on presubmit), but I'd like to get approval on this general approach before going down that rabbit hole of testing.

Reply:
Makes sense. This starts to take us down the road of having multiple implementations also. Possible this could be done as an example instead of a hooked up implementation? I agree showing how this works and how it can be extended makes sense.

Reply:
What do you mean by "example"? 😄